### ISSUE TITLE
unaligned pointer(s) for architecture arm64

### ISSUE THREAD

**Issue by preslavrachev** [go-gl/glfw3#360]
I am getting the following error while trying to run the simple demo code locally:

```
ld: unaligned pointer(s) for architecture arm64
clang-13: error: linker command failed with exit code 1 (use -v to see invocation)
```

I am running Go v1.18.2 on an M1 Mac. I assume that the new chips are not yet fully supported. Is that correct?

**Comment by Jacalz** [comment]
> I assume that the new chips are not yet fully supported. Is that correct?

I don't think that should be the case. We have had https://github.com/fyne-io/fyne working on macOS arm64 for quite a long time now.

**Comment by preslavrachev** [comment]
Looks like this has to do with the Go version. I tried with `gotip`, using the latest Go version:

```
go version devel go1.19-c5da4fb7ac Fri Jul 22 20:12:19 2022 +0000 darwin/arm64
```

and there it worked without any issues. Typical CGO stuff 🤷‍♂️

**Comment by Jacalz** [comment]
Ah, interesting. I see that you mention using Go 1.18.2 but 1.18.4 is the latest. Does it change anything to use the latest stable version?

**Comment by dmitshur** [comment]
`v3.3/glfw` supports `darwin/arm64` as of v3.3.3 (see PR #309) and should work well. I've not had issues on my M1 machine.

Are you running into this problem with `v3.3/glfw` or are you using another version? Can you include more complete steps to reproduce the problem?

**Comment by preslavrachev** [comment]
@Jacalz I've just updated to 1.18.4, and I see the same exact behavior.

@dmitshur That's what I am seeing:
![image](https://user-images.githubusercontent.com/86060/180732630-80491b0c-97b7-4b36-96dc-74783accb540.png)

This is how my `go.mod` looks like:

```go
module glfw-demo

go 1.18

require github.com/go-gl/glfw/v3.3/glfw v0.0.0-20220712193148-63cf1f4ef61f // indirect
```

Anyway, as I mentioned before, it seems to work with the latest version of `gotip` coming form master, so I can do my work, but it is strange nonetheless.

For the record, here are some more version specifics

clang 13.0.1
macOS 12.4

**Comment by Jacalz** [comment]
I'd suggest opening a bug upstream in the `golang/go` repository on GitHub.

FYI, your module seems strange. There should not be any `// indirect` when you are explicitly using it like that. Have you run `go mod tidy`?

**Comment by Jacalz** [comment]
@preslavrachev Is this still an issue with the latest version of this project?

**Comment by Jacalz** [comment]
Closing per https://github.com/go-gl/glfw/issues/360#issuecomment-1193157488, seems like a bug in older versions of the Go compiler. Supposedly fixed in Go 1.19 and later?
